File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        
        String s = String.valueOf(x);
        int number = 0;
        // 각 자리수 더하기
        for(int i=0; i<s.length(); i++) {
            number += s.charAt(i) - '0';
        }
        // x에서 각 자리 더 한수 나눠지면 하샤드
        if(x % number != 0) {
            answer = false;
        }
            
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

bool solution(int x) {
    bool answer = true;
    int number = 0;
    string str = to_string(x);
    
    for(int i=0; i<str.size(); i++) {
        number += str[i] - '0';
    }

    if(x % number == 0) {
        answer = true;
    } else {
        answer = false;
    }
    return answer;
}